### PR TITLE
Add optical pre-emphasis and modify the port name

### PR DESCRIPTION
**Pull request by tiantianlv**
**- What I did**
Add optical pre-emphasis and modify the port name
**- How I did it**
we reserve the EthernetX per lane.  Thus, 2 for 100G and 8 for 400G
**- How to verify it**
root@sonic:/home/admin# show interface status
  Interface                            Lanes    Speed    MTU     Alias    Vlan    Oper    Admin    Type    Asym PFC
-----------  -------------------------------  -------  -----  --------  ------  ------  -------  ------  ----------
  Ethernet0          33,34,35,36,37,38,39,40     400G   9100   QSFPDD1  routed      up       up     N/A         N/A
  Ethernet8          41,42,43,44,45,46,47,48     400G   9100   QSFPDD2  routed    down       up     N/A         N/A
 Ethernet16          49,50,51,52,53,54,55,56     400G   9100   QSFPDD3  routed      up       up     N/A         N/A
 Ethernet24          57,58,59,60,61,62,63,64     400G   9100   QSFPDD4  routed      up       up     N/A         N/A
 Ethernet32          65,66,67,68,69,70,71,72     400G   9100   QSFPDD5  routed      up       up     N/A         N/A
 Ethernet40          73,74,75,76,77,78,79,80     400G   9100   QSFPDD6  routed      up       up     N/A         N/A
 Ethernet48          81,82,83,84,85,86,87,88     400G   9100   QSFPDD7  routed      up       up     N/A         N/A
 Ethernet56          89,90,91,92,93,94,95,96     400G   9100   QSFPDD8  routed      up       up     N/A         N/A
 Ethernet64                  1,2,3,4,5,6,7,8     400G   9100   QSFPDD9  routed      up       up     N/A         N/A
 Ethernet72           9,10,11,12,13,14,15,16     400G   9100  QSFPDD10  routed      up       up     N/A         N/A
 Ethernet80          17,18,19,20,21,22,23,24     400G   9100  QSFPDD11  routed      up       up     N/A         N/A
 Ethernet88          25,26,27,28,29,30,31,32     400G   9100  QSFPDD12  routed      up       up     N/A         N/A
 Ethernet96     97,98,99,100,101,102,103,104     400G   9100  QSFPDD13  routed      up       up     N/A         N/A
Ethernet104  105,106,107,108,109,110,111,112     400G   9100  QSFPDD14  routed    down       up     N/A         N/A
Ethernet112  113,114,115,116,117,118,119,120     400G   9100  QSFPDD15  routed    down       up     N/A         N/A
Ethernet120  121,122,123,124,125,126,127,128     400G   9100  QSFPDD16  routed    down       up     N/A         N/A
Ethernet128  129,130,131,132,133,134,135,136     400G   9100  QSFPDD17  routed    down       up     N/A         N/A
Ethernet136  137,138,139,140,141,142,143,144     400G   9100  QSFPDD18  routed    down       up     N/A         N/A
Ethernet144  145,146,147,148,149,150,151,152     400G   9100  QSFPDD19  routed      up       up     N/A         N/A
Ethernet152  153,154,155,156,157,158,159,160     400G   9100  QSFPDD20  routed      up       up     N/A         N/A
Ethernet160  225,226,227,228,229,230,231,232     400G   9100  QSFPDD21  routed      up       up     N/A         N/A
Ethernet168  233,234,235,236,237,238,239,240     400G   9100  QSFPDD22  routed      up       up     N/A         N/A
Ethernet176  241,242,243,244,245,246,247,248     400G   9100  QSFPDD23  routed      up       up     N/A         N/A
Ethernet184  249,250,251,252,253,254,255,256     400G   9100  QSFPDD24  routed      up       up     N/A         N/A
Ethernet192  161,162,163,164,165,166,167,168     400G   9100  QSFPDD25  routed      up       up     N/A         N/A
Ethernet200  169,170,171,172,173,174,175,176     400G   9100  QSFPDD26  routed      up       up     N/A         N/A
Ethernet208  177,178,179,180,181,182,183,184     400G   9100  QSFPDD27  routed      up       up     N/A         N/A
Ethernet216  185,186,187,188,189,190,191,192     400G   9100  QSFPDD28  routed      up       up     N/A         N/A
Ethernet224  193,194,195,196,197,198,199,200     400G   9100  QSFPDD29  routed      up       up     N/A         N/A
Ethernet232  201,202,203,204,205,206,207,208     400G   9100  QSFPDD30  routed      up       up     N/A         N/A
Ethernet240  209,210,211,212,213,214,215,216     400G   9100  QSFPDD31  routed      up       up     N/A         N/A
Ethernet248  217,218,219,220,221,222,223,224     400G   9100  QSFPDD32  routed      up       up     N/A         N/A
root@sonic:/home/admin#

root@sonic:/home/admin# show interface status
  Interface    Lanes    Speed    MTU     Alias    Vlan    Oper    Admin    Type    Asym PFC
-----------  -------  -------  -----  --------  ------  ------  -------  ------  ----------
  Ethernet0    33,34     100G   9100   QSFP1/1  routed      up       up     N/A         N/A
  Ethernet2    35,36     100G   9100   QSFP1/2  routed      up       up     N/A         N/A
  Ethernet4    37,38     100G   9100   QSFP1/3  routed      up       up     N/A         N/A
  Ethernet6    39,40     100G   9100   QSFP1/4  routed      up       up     N/A         N/A
  Ethernet8    41,42     100G   9100   QSFP2/1  routed    down       up     N/A         N/A
 Ethernet10    43,44     100G   9100   QSFP2/2  routed    down       up     N/A         N/A
 Ethernet12    45,46     100G   9100   QSFP2/3  routed    down       up     N/A         N/A
 Ethernet14    47,48     100G   9100   QSFP2/4  routed      up       up     N/A         N/A
 Ethernet16    49,50     100G   9100   QSFP3/1  routed      up       up     N/A         N/A
 Ethernet18    51,52     100G   9100   QSFP3/2  routed      up       up     N/A         N/A
 Ethernet20    53,54     100G   9100   QSFP3/3  routed      up       up     N/A         N/A
 Ethernet22    55,56     100G   9100   QSFP3/4  routed      up       up     N/A         N/A
 Ethernet24    57,58     100G   9100   QSFP4/1  routed      up       up     N/A         N/A
 Ethernet26    59,60     100G   9100   QSFP4/2  routed      up       up     N/A         N/A
 Ethernet28    61,62     100G   9100   QSFP4/3  routed      up       up     N/A         N/A
 Ethernet30    63,64     100G   9100   QSFP4/4  routed      up       up     N/A         N/A
 Ethernet32    65,66     100G   9100   QSFP5/1  routed      up       up     N/A         N/A
 Ethernet34    67,68     100G   9100   QSFP5/2  routed      up       up     N/A         N/A
 Ethernet36    69,70     100G   9100   QSFP5/3  routed      up       up     N/A         N/A
 Ethernet38    71,72     100G   9100   QSFP5/4  routed      up       up     N/A         N/A
 Ethernet40    73,74     100G   9100   QSFP6/1  routed    down       up     N/A         N/A
 Ethernet42    75,76     100G   9100   QSFP6/2  routed      up       up     N/A         N/A
 Ethernet44    77,78     100G   9100   QSFP6/3  routed      up       up     N/A         N/A
 Ethernet46    79,80     100G   9100   QSFP6/4  routed      up       up     N/A         N/A
 Ethernet48    81,82     100G   9100   QSFP7/1  routed      up       up     N/A         N/A
 Ethernet50    83,84     100G   9100   QSFP7/2  routed      up       up     N/A         N/A
 Ethernet52    85,86     100G   9100   QSFP7/3  routed      up       up     N/A         N/A
 Ethernet54    87,88     100G   9100   QSFP7/4  routed      up       up     N/A         N/A
 Ethernet56    89,90     100G   9100   QSFP8/1  routed      up       up     N/A         N/A
 Ethernet58    91,92     100G   9100   QSFP8/2  routed      up       up     N/A         N/A
 Ethernet60    93,94     100G   9100   QSFP8/3  routed      up       up     N/A         N/A
 Ethernet62    95,96     100G   9100   QSFP8/4  routed      up       up     N/A         N/A
 Ethernet64      1,2     100G   9100   QSFP9/1  routed      up       up     N/A         N/A
 Ethernet66      3,4     100G   9100   QSFP9/2  routed      up       up     N/A         N/A
 Ethernet68      5,6     100G   9100   QSFP9/3  routed      up       up     N/A         N/A
 Ethernet70      7,8     100G   9100   QSFP9/4  routed      up       up     N/A         N/A
 Ethernet72     9,10     100G   9100  QSFP10/1  routed      up       up     N/A         N/A
 Ethernet74    11,12     100G   9100  QSFP10/2  routed      up       up     N/A         N/A
 Ethernet76    13,14     100G   9100  QSFP10/3  routed      up       up     N/A         N/A
 Ethernet78    15,16     100G   9100  QSFP10/4  routed      up       up     N/A         N/A
 Ethernet80    17,18     100G   9100  QSFP11/1  routed      up       up     N/A         N/A
 Ethernet82    19,20     100G   9100  QSFP11/2  routed      up       up     N/A         N/A
 Ethernet84    21,22     100G   9100  QSFP11/3  routed      up       up     N/A         N/A
 Ethernet86    23,24     100G   9100  QSFP11/4  routed      up       up     N/A         N/A
 Ethernet88    25,26     100G   9100  QSFP12/1  routed      up       up     N/A         N/A
 Ethernet90    27,28     100G   9100  QSFP12/2  routed      up       up     N/A         N/A
 Ethernet92    29,30     100G   9100  QSFP12/3  routed      up       up     N/A         N/A
 Ethernet94    31,32     100G   9100  QSFP12/4  routed      up       up     N/A         N/A
 Ethernet96    97,98     100G   9100  QSFP13/1  routed      up       up     N/A         N/A
 Ethernet98   99,100     100G   9100  QSFP13/2  routed      up       up     N/A         N/A
Ethernet100  101,102     100G   9100  QSFP13/3  routed      up       up     N/A         N/A
Ethernet102  103,104     100G   9100  QSFP13/4  routed      up       up     N/A         N/A
Ethernet104  105,106     100G   9100  QSFP14/1  routed      up       up     N/A         N/A
Ethernet106  107,108     100G   9100  QSFP14/2  routed    down       up     N/A         N/A
Ethernet108  109,110     100G   9100  QSFP14/3  routed    down       up     N/A         N/A
Ethernet110  111,112     100G   9100  QSFP14/4  routed    down       up     N/A         N/A
Ethernet112  113,114     100G   9100  QSFP15/1  routed    down       up     N/A         N/A
Ethernet114  115,116     100G   9100  QSFP15/2  routed    down       up     N/A         N/A
Ethernet116  117,118     100G   9100  QSFP15/3  routed    down       up     N/A         N/A
Ethernet118  119,120     100G   9100  QSFP15/4  routed    down       up     N/A         N/A
Ethernet120  121,122     100G   9100  QSFP16/1  routed    down       up     N/A         N/A
Ethernet122  123,124     100G   9100  QSFP16/2  routed    down       up     N/A         N/A
Ethernet124  125,126     100G   9100  QSFP16/3  routed    down       up     N/A         N/A
Ethernet126  127,128     100G   9100  QSFP16/4  routed    down       up     N/A         N/A
Ethernet128  129,130     100G   9100  QSFP17/1  routed    down       up     N/A         N/A
Ethernet130  131,132     100G   9100  QSFP17/2  routed    down       up     N/A         N/A
Ethernet132  133,134     100G   9100  QSFP17/3  routed    down       up     N/A         N/A
Ethernet134  135,136     100G   9100  QSFP17/4  routed    down       up     N/A         N/A
Ethernet136  137,138     100G   9100  QSFP18/1  routed    down       up     N/A         N/A
Ethernet138  139,140     100G   9100  QSFP18/2  routed    down       up     N/A         N/A
Ethernet140  141,142     100G   9100  QSFP18/3  routed    down       up     N/A         N/A
Ethernet142  143,144     100G   9100  QSFP18/4  routed    down       up     N/A         N/A
Ethernet144  145,146     100G   9100  QSFP19/1  routed      up       up     N/A         N/A
Ethernet146  147,148     100G   9100  QSFP19/2  routed    down       up     N/A         N/A
Ethernet148  149,150     100G   9100  QSFP19/3  routed    down       up     N/A         N/A
Ethernet150  151,152     100G   9100  QSFP19/4  routed    down       up     N/A         N/A
Ethernet152  153,154     100G   9100  QSFP20/1  routed      up       up     N/A         N/A
Ethernet154  155,156     100G   9100  QSFP20/2  routed      up       up     N/A         N/A
Ethernet156  157,158     100G   9100  QSFP20/3  routed      up       up     N/A         N/A
Ethernet158  159,160     100G   9100  QSFP20/4  routed      up       up     N/A         N/A
Ethernet160  225,226     100G   9100  QSFP21/1  routed      up       up     N/A         N/A
Ethernet162  227,228     100G   9100  QSFP21/2  routed      up       up     N/A         N/A
Ethernet164  229,230     100G   9100  QSFP21/3  routed      up       up     N/A         N/A
Ethernet166  231,232     100G   9100  QSFP21/4  routed      up       up     N/A         N/A
Ethernet168  233,234     100G   9100  QSFP22/1  routed      up       up     N/A         N/A
Ethernet170  235,236     100G   9100  QSFP22/2  routed      up       up     N/A         N/A
Ethernet172  237,238     100G   9100  QSFP22/3  routed      up       up     N/A         N/A
Ethernet174  239,240     100G   9100  QSFP22/4  routed      up       up     N/A         N/A
Ethernet176  241,242     100G   9100  QSFP23/1  routed      up       up     N/A         N/A
Ethernet178  243,244     100G   9100  QSFP23/2  routed      up       up     N/A         N/A
Ethernet180  245,246     100G   9100  QSFP23/3  routed      up       up     N/A         N/A
Ethernet182  247,248     100G   9100  QSFP23/4  routed      up       up     N/A         N/A
Ethernet184  249,250     100G   9100  QSFP24/1  routed      up       up     N/A         N/A
Ethernet186  251,252     100G   9100  QSFP24/2  routed      up       up     N/A         N/A
Ethernet188  253,254     100G   9100  QSFP24/3  routed      up       up     N/A         N/A
Ethernet190  255,256     100G   9100  QSFP24/4  routed      up       up     N/A         N/A
Ethernet192  161,162     100G   9100  QSFP25/1  routed      up       up     N/A         N/A
Ethernet194  163,164     100G   9100  QSFP25/2  routed      up       up     N/A         N/A
Ethernet196  165,166     100G   9100  QSFP25/3  routed      up       up     N/A         N/A
Ethernet198  167,168     100G   9100  QSFP25/4  routed      up       up     N/A         N/A
Ethernet200  169,170     100G   9100  QSFP26/1  routed      up       up     N/A         N/A
Ethernet202  171,172     100G   9100  QSFP26/2  routed      up       up     N/A         N/A
Ethernet204  173,174     100G   9100  QSFP26/3  routed      up       up     N/A         N/A
Ethernet206  175,176     100G   9100  QSFP26/4  routed      up       up     N/A         N/A
Ethernet208  177,178     100G   9100  QSFP27/1  routed      up       up     N/A         N/A
Ethernet210  179,180     100G   9100  QSFP27/2  routed      up       up     N/A         N/A
Ethernet212  181,182     100G   9100  QSFP27/3  routed      up       up     N/A         N/A
Ethernet214  183,184     100G   9100  QSFP27/4  routed      up       up     N/A         N/A
Ethernet216  185,186     100G   9100  QSFP28/1  routed      up       up     N/A         N/A
Ethernet218  187,188     100G   9100  QSFP28/2  routed      up       up     N/A         N/A
Ethernet220  189,190     100G   9100  QSFP28/3  routed      up       up     N/A         N/A
Ethernet222  191,192     100G   9100  QSFP28/4  routed      up       up     N/A         N/A
Ethernet224  193,194     100G   9100  QSFP29/1  routed      up       up     N/A         N/A
Ethernet226  195,196     100G   9100  QSFP29/2  routed      up       up     N/A         N/A
Ethernet228  197,198     100G   9100  QSFP29/3  routed      up       up     N/A         N/A
Ethernet230  199,200     100G   9100  QSFP29/4  routed      up       up     N/A         N/A
Ethernet232  201,202     100G   9100  QSFP30/1  routed      up       up     N/A         N/A
Ethernet234  203,204     100G   9100  QSFP30/2  routed      up       up     N/A         N/A
Ethernet236  205,206     100G   9100  QSFP30/3  routed      up       up     N/A         N/A
Ethernet238  207,208     100G   9100  QSFP30/4  routed      up       up     N/A         N/A
Ethernet240  209,210     100G   9100  QSFP31/1  routed      up       up     N/A         N/A
Ethernet242  211,212     100G   9100  QSFP31/2  routed      up       up     N/A         N/A
Ethernet244  213,214     100G   9100  QSFP31/3  routed      up       up     N/A         N/A
Ethernet246  215,216     100G   9100  QSFP31/4  routed      up       up     N/A         N/A
Ethernet248  217,218     100G   9100  QSFP32/1  routed      up       up     N/A         N/A
Ethernet250  219,220     100G   9100  QSFP32/2  routed      up       up     N/A         N/A
Ethernet252  221,222     100G   9100  QSFP32/3  routed      up       up     N/A         N/A
Ethernet254  223,224     100G   9100  QSFP32/4  routed      up       up     N/A         N/A
root@sonic:/home/admin#

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: device/celestica/x86_64-cel_silverstone-r0/Silverstone-128x100/pre_emphasis_128x100G_PAM4_optics.soc [hunk 1]
```diff
@@ -0,0 +1,2454 @@
+
```
[the remark this answers is not in the document]
Comment (tiantianlv) [reply device/celestica/x86_64-cel_silverstone-r0/Silverstone-128x100/pre_emphasis_128x100G_PAM4_optics.soc]: add handle in led_proc_init.soc

File: device/celestica/x86_64-cel_silverstone-r0/Silverstone-128x100/pre-emphasis_PAM4_optics.soc [hunk 2]
```diff
@@ -0,0 +1,2454 @@
+
+#***
+#*** Port ce0 Preemphasis setting ***
+#***
+
+local port ce0
+#*** lane 0 ***
+phy $port TXFIR_TAP_CTL0r.0 TXFIR_TAP0_COEFF=0
```
Comment (lguohan) [inline device/celestica/x86_64-cel_silverstone-r0/Silverstone-128x100/pre-emphasis_PAM4_optics.soc]: using config.bcm?

Comment (tiantianlv) [reply device/celestica/x86_64-cel_silverstone-r0/Silverstone-128x100/pre-emphasis_PAM4_optics.soc]: Broadcom FAE told me only the pre-emphasis of port was support in config,bcm for BH serdes, not for lane. But our SI team only gets the pre-emphasis of lane for BH serdes. so I have to use SOC file to handle it.
Thanks,
Bo

**Comment by zhenggen-xu** [comment]
Please backport to 201811 branch once https://github.com/Azure/sonic-buildimage/pull/3678 was merged.

**Pull request closed without merging**